go/src/koding/klientctl/auth/register.go

```diff
@@ -0,0 +1,206 @@
+package auth
+
+import (
+	"encoding/json"
+	"fmt"
+	"koding/httputil"
+	"koding/klientctl/config"
+	"koding/klientctl/ctlcli"
+	"koding/klientctl/kloud"
+	"net/http"
+	"net/http/cookiejar"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/codegangsta/cli"
+	"github.com/koding/logging"
+)
+
+// NewRegisterSubCommand provides the subcommand for registering
+func NewRegisterSubCommand(log logging.Logger) cli.Command {
+	return cli.Command{
+		Name:   "register",
+		Usage:  "Registers the user",
+		Action: ctlcli.ExitAction(RegisterCommand, log, "register"),
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:  "username",
```

Comment:
>  registering with default without --username name ?

Instead of taking values from flags, ask user on stdin, like `kd credential create`: http://recordit.co/UwA6e8JjS8